code/modules/projectiles/guns/energy.dm

```diff
@@ -29,69 +30,150 @@
 	else
 		power_supply = new(src)
 	power_supply.give(power_supply.maxcharge)
-	if(self_recharge)
+	var/obj/item/ammo_casing/energy/shot
+	for (var/i = 1, i <= ammo_type.len, i++)
+		var/shottype = ammo_type[i]
+		shot = new shottype(src)
+		ammo_type[i] = shot
+	shot = ammo_type[select]
+	fire_sound = shot.fire_sound
+	fire_delay = shot.delay
+	if(selfcharge)
 		processing_objects.Add(src)
 	update_icon()
 
 /obj/item/weapon/gun/energy/Destroy()
-	if(self_recharge)
+	if(selfcharge)
 		processing_objects.Remove(src)
 	return ..()
 
 /obj/item/weapon/gun/energy/process()
-	if(self_recharge) //Every [recharge_time] ticks, recharge a shot for the cyborg
+	if(selfcharge) //Every [recharge_time] ticks, recharge a shot for the cyborg
 		charge_tick++
-		if(charge_tick < recharge_time) return 0
+		if(charge_tick < charge_delay)
+			return 0

```

Comment:
these are regular `return`'s on TG, would it be applicable to keep them `return 0` or should they be just regular `return`?